### ISSUE TITLE
[Vue][Bug] Long v-if condition break the code after reformat

### ISSUE THREAD

**Issue by maguszoldik** [prettier/prettier#5561]
**Prettier 1.15.2**
**Vue 2.5.17**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWWa4FxmnPMDOOEMALzEQAJwgB3APpwAHljxQqcUSK7cyfAQFoekLHGEhg4iQF81TbnW00AZkf5QA5gGFotGDWgBGMgDJ-MkUBAAsVABloNw8aL2gyQSSyAHIMAE9giTgeCAwOURyAVxwYFICgkJhw0QB1UJowUIAhQrwAawAVcIiaKDgIhGdqkUYNSlz86r7nMgB6MY1USbhplwX1ClQ51nwidW3aBlJt3fZGEAAaEAgseKgeZFA8UVMABReER5Q8OggaKhXEAAI1EeDA7VWAGUFGAZsgYKIinBrqEYBgcPU4jlYXAoV84vQ4ulkOAeI9rn0eCoYG8wc4MHhkHY8DhqdcAFY8WStcGQmBQvD5Xr9Zms9kgLmyKEzHBwACKRQEcDFbJRIAUompolJdGRQKwoj6MFqAOqyAAHAAGa6GiDU2pgrCkw05FR0FXXQoARyKNEKdLwDKZSBZauu1IwNARSPVPFlCqV8FVEpgeGBpqo5qQACZroi8DQcDN3BhGaSoNBPSAitTOunvmHqWYzEA)
```sh
# Options (if any):
--parser vue
```

**Input:**
A `v-if` is used on a `template` element which is also declared as a `slot`.

```jsx
<template>
  <div>
    <template 
      slot="row_expander"
      slot-scope="{row}"
      v-if="longCondition1 && anotherLongCondition === 'my awesome result' && anotherWhichBreakTheLineLength">
      <something />
      <something/>
    </template>
  </div>
</template>
```

**Output:**
```jsx
<template>
  <div>
    <template
      slot="row_expander"
      slot-scope="{
        row
      }"
      v-if="
        longCondition1 &&
          anotherLongCondition === 'my awesome result' &&
          anotherWhichBreakTheLineLength
      "
    >
      <something /> <something />
    </template>
  </div>
</template>
```
In this particular case (usage on template element which is a slot renderer), the resulting code will break.

**Expected behavior:**
The code must be encapsulated in parenthesis to properly work. (And indentation may be better imho.)

```jsx
<template>
  <div>
    <template
      slot="row_expander"
      slot-scope="{
        row
      }"
      v-if="(
        longCondition1 &&
        anotherLongCondition === 'my awesome result' &&
        anotherWhichBreakTheLineLength
     ) "
    >
      <something /> <something />
    </template>
  </div>
</template>
```

**Comment by ikatyang** [comment]
I'm confusing, how did it break your code? I tried it on [Vue Template Explorer](https://template-explorer.vuejs.org) and I didn't see anything changed: 

- ```vue
  <div v-if="longCondition1 && anotherLongCondition === 'my awesome result' && anotherWhichBreakTheLineLength">
    Templates are not formatted yet ...
  </div>
  ```
  ```js
  function render() {
    with(this) {
      return (longCondition1 && anotherLongCondition === 'my awesome result' &&
        anotherWhichBreakTheLineLength) ? _c('div', [_v(
        "\n    Templates are not formatted yet ...\n  ")]) : _e()
    }
  }
  ```
- ```vue
  <div
    v-if="
      longCondition1 &&
        anotherLongCondition === 'my awesome result' &&
        anotherWhichBreakTheLineLength
    "
  >
    Templates are not formatted yet ...
  </div>
  ```
  ```js
  function render() {
    with(this) {
      return (
          longCondition1 &&
          anotherLongCondition === 'my awesome result' &&
          anotherWhichBreakTheLineLength
        ) ? _c('div', [_v("\n    Templates are not formatted yet ...\n  ")]) :
        _e()
    }
  }
  ```

**Comment by maguszoldik** [comment]
My bad, the real code was more complex and in fact it seems to be an edge case of the template compiler.
It break the code when the element is a `template` and is used to render a `slot`. See [this updated example](https://template-explorer.vuejs.org/#%3Cdiv%3E%0A%20%20%20%20%3Ctemplate%0A%20%20%20%20%20%20slot%3D%22row_expander%22%0A%20%20%20%20%20%20slot-scope%3D%22%7B%0A%20%20%20%20%20%20%20%20row%0A%20%20%20%20%20%20%7D%22%0A%20%20%20%20%20%20v-if%3D%22%0A%20%20%20%20%20%20%20%20longCondition1%20%26%26%0A%20%20%20%20%20%20%20%20%20%20anotherLongCondition%20%3D%3D%3D%20'my%20awesome%20result'%20%26%26%0A%20%20%20%20%20%20%20%20%20%20anotherWhichBreakTheLineLength%0A%20%20%20%20%20%20%22%0A%20%20%20%20%3E%0A%20%20%20%20%20%20%3Csomething%20%2F%3E%20%3Csomething%20%2F%3E%0A%20%20%20%20%3C%2Ftemplate%3E%0A%20%3C%2Fdiv%3E)

I edited the initial report.

**Comment by ikatyang** [comment]
This looks like a bug in Vue, can you open an issue there?

**Comment by ikatyang** [comment]
As a workaround, you could use [`<!-- prettier-ignore-attribute v-if -->`](https://prettier.io/docs/en/ignore.html#html):

**Prettier 1.15.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWUCEAtG2VgE5www043NjQDmUCLzYEY3GgCMArvDJ0RAMzIdGzSplwE4ZUrooBnHBBgBeYiG4QA7gH04ADyx4oVIXZOmLKzYzSCw4WxBgB0cAXz8mXTUadQjLKFEAYWhaAWgARjIAMkKyLysACyEAGWhM7Jpc8mtmsgByDABPUsc4MwgMI14zRRwYVqKSsphK7gB1cpowcoAhXjwAawAVSqqaKDgqhFFpux1TVD6B6b3RMgB6M90L-r4F9If-ClQ7-Xwif2+tAYpG+v0MjBAABoQBAsI0zMhQHhuNEAArIhAIlB4OgQGhUKEgeTcPBgdZ8ADKnjAN2QskUcGh5RgGBw8wavWpcApmIa9AaHWQ4DMCOhezMQhgqJJogweGQ6jwOAl0IAVmY3KtSeSYBS8ANdvsFUqVSB1W4KTccHAAIqKKxwY3KxkgTzcCXcIV0BmEnh7GCzfHTZAADgADNCeBAJbMSVghTxekI6I7obwAI6KGi8aV4WXypCK53QiUYGh07gMktW232+BO00wPDyQNUYNIABM0NkeBoOBuWQwcqFEiN0MUEs2zaxRYlMRiQA)
```sh
--parser vue
```

**Input:**
```vue
<template>
  <div>
    <!-- prettier-ignore-attribute v-if -->
    <template 
      slot="row_expander"
      slot-scope="{row}"
      v-if="longCondition1 && anotherLongCondition === 'my awesome result' && anotherWhichBreakTheLineLength">
      <something />
      <something/>
    </template>
  </div>
</template>

```

**Output:**
```vue
<template>
  <div>
    <!-- prettier-ignore-attribute v-if -->
    <template
      slot="row_expander"
      slot-scope="{
        row
      }"
      v-if="longCondition1 && anotherLongCondition === 'my awesome result' && anotherWhichBreakTheLineLength"
    >
      <something /> <something />
    </template>
  </div>
</template>

```

**Comment by ikatyang** [comment]
Fixed in vuejs/vue#9119.
